src/main/java/EntropySolver.java

```diff
@@ -0,0 +1,129 @@
+            ACCEPTED.forEach(potentialInput -> {
+                Map<Integer, Integer> patterns = newHashMap();
+                solutions.forEach(solution -> patterns.merge(calculatePattern(potentialInput, solution), 1, Integer::sum));
+                entropyMap.put(potentialInput, getEntropy(patterns));
+            });
+
+            String bestWord = getBestWord(entropyMap);
+            log.info("Best Word: {}", bestWord);
+            filterRemainingWords(solutions, bestWord, getOutcome());
+            log.info("{} - {}", solutions.size(), solutions.size() < 100 ? solutions : "");
+        }
+        log.info("Solution: {}", solutions.iterator().next());
+    }
+
+    private static String getBestWord(Map<String, Double> entropyMap) {
+        Map<String, Double> sortedMap = newLinkedHashMap();
+        entropyMap.entrySet()
+                .stream()
+                .sorted((o1, o2) -> o2.getValue()
+                        .compareTo(o1.getValue()))
+                .forEach(entry -> sortedMap.put(entry.getKey(), entry.getValue()));
+
+        return sortedMap.entrySet()
+                .iterator()
+                .next()
+                .getKey();
+    }
+
+    private static Double getEntropy(Map<Integer, Integer> patterns) {
+        return patterns.values()
+                .stream()
+                .mapToDouble(pattern -> pattern / (ACCEPTED.size() * 1.0))
+                .map(probability -> probability * log(1 / probability))
+                .sum();
+    }
+
+    static int calculatePattern(String word1, String word2) {
+        int pattern = 0;
+        for (int i = 0; i < word1.length(); i++) {
+            double pow = Math.pow(3, (word1.length() - i - 1));
+            // Green
+            if (word1.charAt(i) == word2.charAt(i)) {
+                pattern += (2 * pow);
+            } else if (word2.contains(word1.charAt(i) + "")) {
+                pattern += pow;
+            }
+        }
+        return pattern;
+    }
+
+    private static void filterRemainingWords(Set<String> solutions, String wordUsed, String outcome) {
+        solutions.removeIf(word -> !isWordStillValid(word, wordUsed, outcome));
+    }
+
+    private static boolean isWordStillValid(String currentWord, String wordUsed, String outcome) {
+        for (int i = 0; i < currentWord.length(); i++) {
+            char used = wordUsed.charAt(i);
+            char current = currentWord.charAt(i);
+            boolean currentContainsUsed = currentWord.contains(used + "");
+            if (outcome.charAt(i) == 'x' && currentContainsUsed && (wordUsed + "!").split(used + "").length == 2) {
+                return false;
+            }
+            if (outcome.charAt(i) == 'y' && (!currentContainsUsed || current == used)) {
+                return false;
+            }
+            if (outcome.charAt(i) == 'g' && current != used) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @SneakyThrows
```

Comment:
Blue - Why are you using `@SneakyThrows`. Would it hide critical errors if System.in has issues